File: client_sdks/devconnect_flutter/lib/src/reporters/storage_reporter.dart
import '../devconnect_client.dart';

/// Reports storage operations to DevConnect.
///
/// Usage:
/// ```dart
/// final storage = DevConnectStorage(storageType: 'shared_preferences');
/// storage.reportWrite('user_token', 'abc123');
/// storage.reportRead('user_token', 'abc123');
/// storage.reportDelete('user_token');
/// ```
class DevConnectStorage {
  final String storageType;

  const DevConnectStorage({this.storageType = 'shared_preferences'});

  void reportRead(String key, dynamic value) {
    DevConnectClient.safeReportStorageOperation(
      storageType: storageType,
      key: key,
      value: value,
      operation: 'read',
    );
  }

  void reportWrite(String key, dynamic value) {
    DevConnectClient.safeReportStorageOperation(
      storageType: storageType,
      key: key,
      value: value,
      operation: 'write',
    );
  }

  void reportDelete(String key) {
    DevConnectClient.safeReportStorageOperation(
      storageType: storageType,
      key: key,
      operation: 'delete',
    );
  }

  void reportClear() {
    DevConnectClient.safeReportStorageOperation(
      storageType: storageType,
      key: '*',
      operation: 'clear',
    );
  }
}

File: client_sdks/devconnect-react-native/src/reporters/storageReporter.ts
import { DevConnect } from '../client';

/**
 * Manual storage reporter for non-AsyncStorage stores (MMKV, SecureStore, etc.)
 *
 * ```typescript
 * const mmkvReporter = new DevConnectStorage('mmkv');
 * mmkvReporter.reportWrite('user_token', 'abc123');
 * ```
 */
export class DevConnectStorage {
  constructor(private storageType: string = 'async_storage') {}

  reportRead(key: string, value?: any): void {
    DevConnect.reportStorageOperation({
      storageType: this.storageType, key, value, operation: 'read',
    });
  }

  reportWrite(key: string, value?: any): void {
    DevConnect.reportStorageOperation({
      storageType: this.storageType, key, value, operation: 'write',
    });
  }

  reportDelete(key: string): void {
    DevConnect.reportStorageOperation({
      storageType: this.storageType, key, operation: 'delete',
    });
  }

  reportClear(): void {
    DevConnect.reportStorageOperation({
      storageType: this.storageType, key: '*', operation: 'clear',
    });
  }
}
